### DISCUSSION TITLE
Mimesis data fields

### DISCUSSION THREAD

**Discussion by abhdhar-1618** [lk-geimfari/mimesis#1449] [Q&A] [answered]
Hello everyone,

I am trying to build a synthetic data set for my own work in data analysis. However, I am not sure how many data fields the module actually provides. So, can I have a complete list of data fields that can be generated by the module and a way the fetch that data (mainly in python)? Thanks!

**Comment by lk-geimfari** [comment, marked as answer]
Hi. There is no a complete list of supported fields, but you can see methods of each data provider here:

- [`Locale-Dependend Providers`](https://mimesis.name/en/master/api.html#locale-dependent-providers)
- [`Locale-Independent Providers`](https://mimesis.name/en/master/api.html#locale-independent-providers)

Also it's good to understand how `fields` actually works: 

- [`Structured Data Generation`](https://mimesis.name/en/master/schema.html#schema-and-field)

**Reply by abhdhar-1618** [reply]
Thanks for the help. I was able to pull the Locale-Dependent Providers but not able to pull the Locale-indipendant providers.
